### PR TITLE
Improve system tests and remove_all logic

### PR DESCRIPTION
**Pull request by lagalbra**
No extraneous messages are printed out during system testing
Improvement of remove_all workspaces logic for non-OMS workspace cases
Localized unit test for containerized agent hostname

@Microsoft/omsagent-devs

### CODE REVIEW

**Comment by lagalbra** [comment]
pBuild running now

**Comment by lagalbra** [comment]
pBuild, unit tests, and system tests pass.

**Comment by lagalbra** [comment]
@v-xecamp Please review. This should be checked in before https://github.com/Microsoft/OMS-Agent-for-Linux/pull/575

File: installer/scripts/omsadmin.sh [hunk 1]
```diff
@@ -592,6 +595,7 @@ onboard()
         echo
         echo "Generated request:"
         cat $BODY_ONBOARD
+        echo
```
Comment (v-xecamp) [inline installer/scripts/omsadmin.sh]: I could completely change this to a here document as it is much more readable and maintainable:
cat <<EOGREQ # or whatever

Generated request:
$BODY_ONBOARD

EOGREQ

File: installer/scripts/omsadmin.sh [hunk 2]
```diff
@@ -632,7 +636,7 @@ onboard()
         cleanup_certs
         return $INVALID_PROXY
     elif [ $error -ne 0 ]; then
-        log_error "Error during the onboarding request. Check the correctness of the workspace ID and shared key or run omsadmin.sh with '-v'"
+        log_error "Error during the onboarding request: curl returned $error. Check the correctness of the workspace ID and shared key or run omsadmin.sh with '-v'"
         cleanup_certs
         return $ERROR_ONBOARDING
```
Comment (v-xecamp) [inline installer/scripts/omsadmin.sh]: I thought I saw somewhere that there are curl specific errors.  Are those pertinent here?  Perhaps even to embed in the log message if not to return.

File: test/installer/scripts/maintenance_systestbase.rb [hunk 3]
```diff
@@ -72,16 +72,26 @@ def prep_proxy(proxy_setting)
     check_proxy_server_working
   end
 
-  def do_onboard(workspace_id, shared_key, should_succeed = true)
+  def do_onboard(workspace_id, shared_key, should_succeed = true, verbose = false)
     check_test_keys()
-    onboard_out = `#{@omsadmin_script} -w #{workspace_id} -s #{shared_key}`
-    result = $?
+    onboard_cmd = "#{@omsadmin_script}"
+    if !workspace_id.nil?
```
Comment (v-xecamp) [inline test/installer/scripts/maintenance_systestbase.rb]: unless workspace_id.nil?

File: test/installer/scripts/maintenance_systestbase.rb [hunk 4]
```diff
@@ -72,16 +72,26 @@ def prep_proxy(proxy_setting)
...
+       onboard_cmd = "#{onboard_cmd} -w #{workspace_id}"
+    end
+    if !shared_key.nil?
```
Comment (v-xecamp) [inline test/installer/scripts/maintenance_systestbase.rb]: unless shared_key.nil?

File: installer/scripts/omsadmin.sh (hunk 1 above)
Comment (lagalbra) [reply installer/scripts/omsadmin.sh]: As you know, I'm changing how the generated request is printed in https://github.com/Microsoft/OMS-Agent-for-Linux/pull/575, and I don't think this comment applies once those changes are taken into consideration.

File: installer/scripts/omsadmin.sh (hunk 2 above)
Comment (lagalbra) [reply installer/scripts/omsadmin.sh]: You can see the curl-specific error codes here: https://help.github.com/articles/testing-your-ssh-connection/

If you look in the lines of code above these, we check if any of those curl-specific error codes apply; if not, we fall to this case. This changed line is just adding curl's return code to the printout for diagnostic purposes.

File: test/installer/scripts/maintenance_systestbase.rb (hunk 3 above)
Comment (lagalbra) [reply test/installer/scripts/maintenance_systestbase.rb]: Will change

File: test/installer/scripts/maintenance_systestbase.rb (hunk 4 above)
Comment (lagalbra) [reply test/installer/scripts/maintenance_systestbase.rb]: Will change

**Pull request merged**